### DISCUSSION TITLE
Is it possible to add an older version of vagrant to an existing tap?

### DISCUSSION THREAD

**Discussion by DevScissors** [Homebrew/discussions#2982] [Tap maintenance and Homebrew development] [answered]
At my company we use vagrant and virtual box but our devops is quite behind (still on python 2.6). With that being said, anything past vagrant 2.2.9 breaks and doesn't work. I used to install it through brew with the specific version in the cask but that has since been deprecated. I noticed there is now a tap for hashicorp but it only has the latest version available. The interesting thing is under releases (https://releases.hashicorp.com/vagrant/2.2.9/) the version exists. 

So is it possible to make a pr for their github to add that to the tap? Is that even the process to update/add to a tap?

I have used brew for awhile but never worked with actually adding packages to brew myself.

**Comment by carlocab** [comment, marked as answer]
The [Hashicorp tap](https://github.com/hashicorp/homebrew-tap) is run by Hashicorp, so you'll need to ask them.

Note that anyone can create a tap, so even if Hashicorp don't want to host the formula you need, you can always create your own. See:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

**Reply by DevScissors** [reply]
Wow, such a fast response! So I can create my own tap and add their source code to then install that specific version through brew? This is quite intriguing and I think I have a new project to start :)

**Reply by carlocab** [reply]
Yes. Since it's your own tap, you can add whatever you want to it--not just alternate versions of formulae you've found elsewhere. We often suggest this a way of using Homebrew to install software that don't meet the requirements for inclusion in homebrew-core, for example.
